src/filters/FromXMLFilter.C

```diff
@@ -0,0 +1,98 @@
+/********************************************************************/
+/*                  SOFTWARE COPYRIGHT NOTIFICATION                 */
+/*                             Cardinal                             */
+/*                                                                  */
+/*                  (c) 2021 UChicago Argonne, LLC                  */
+/*                        ALL RIGHTS RESERVED                       */
+/*                                                                  */
+/*                 Prepared by UChicago Argonne, LLC                */
+/*               Under Contract No. DE-AC02-06CH11357               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*             Prepared by Battelle Energy Alliance, LLC            */
+/*               Under Contract No. DE-AC07-05ID14517               */
+/*                With the U. S. Department of Energy               */
+/*                                                                  */
+/*                 See LICENSE for full restrictions                */
+/********************************************************************/
+
+#ifdef ENABLE_OPENMC_COUPLING
+
+#include "FromXMLFilter.h"
+
+registerMooseObject("CardinalApp", FromXMLFilter);
+
+InputParameters
+FromXMLFilter::validParams()
+{
+  auto params = FilterBase::validParams();
+  params.addClassDescription(
+      "A class which provides a thin wrapper around an arbitrary OpenMC filter.");
+  params.addRequiredRangeCheckedParam<int>(
+      "filter_id",
+      "filter_id >= 0",
+      "The id of the OpenMC filter that this class should provide to Cardinal tallies.");
+  params.addRequiredParam<std::string>("bin_label",
+                                       "The label that is used for this filter's bins.");
+  params.addParam<bool>("allow_expansion_filters",
+                        false,
+                        "Whether functional expansion filters are allowed or not. Tallies with "
+                        "these filters are likely to fail "
+                        "automatic normalization as a sum over all bins may not make sense for a "
+                        "certain functional expansion.");
+
+  return params;
+}
+
+FromXMLFilter::FromXMLFilter(const InputParameters & parameters)
```

Comment:
We should probably throw an error if the user is trying to add a spatial filter, right? Since we're handling those via Cardinal?